docs/howto/UPGRADING.md

```diff
@@ -0,0 +1,93 @@
+# How to upgrade from previous versions
+
+### Creating a Custom Base Adapter
+
+You can upgrade to ams ``.10`` by creating a custom Base Adapter to
+emulate the response that your application currently serves.
+
+###### Base Adapter
+
+A base adapter takes a ``ActiveModel::Serializer`` or
+``ActiveModel::Serializer::ArraySerializer`` instance, and creates a hash
+used for serialization in its ``serializable_hash`` method. A base adapter
+could look like this:
+
+```ruby
+class MyApp::BaseAdapter < ActiveModel::Serializer::Adapter::Base
```

Comment:
typo: `;:`.  why not try adding this code with tests? 

I'm thinking

```ruby
# lib/active_model_serializers/v08/serializer.rb
# lib/active_model_serializers/v08/collection_serializer.rb
# lib/active_model_serializers/v08/adapter.rb
module ActiveModelSerializers::V08
  # Serializer might not need anything, but good to have defined
  Serializer = ActiveModel::Serializer
  CollectionSerializer = ActiveModel::Serializer::CollectionSerializer # well, it's Array now, but hopefully we'll get that renaming pr soon
  class Adapter < ActiveModel::Serializer::Adapter::Base
    # whatever methods you need
  end
end
```

And test like the other serializers and adapters

```ruby
adapter = :'v08/adapter'
serializer = :'v08/array_serializer'
each_serializer = :'v08/serializer'
collection_resource = Post.all
expected_response = [{ something }]
assert ActiveModel::SerializableResource.new(collection_resource, serializer: serializer, each_serializer: each_serializer, adapter: adapter).as_json, expected_response

# and

adapter = :'v08/adapter'
serializer = :'v08/serializer'
resource = Post.first
expected_response = { something }
assert ActiveModel::SerializableResource.new(collection_resource, serializer: serializer, adapter: adapter).as_json, expected_response
```